Fix tests run without GPU

```
FAILED tests/test_array_utils.py::test_has_usm_memory[opencl:gpu:0] - dpctl._...
FAILED tests/test_array_utils.py::test_as_usm_obj[opencl:gpu:0] - dpctl._sycl...
FAILED tests/test_array_utils.py::test_has_usm_memory[level_zero:gpu:0] - dpc...
FAILED tests/test_array_utils.py::test_as_usm_obj[level_zero:gpu:0] - dpctl._...
FAILED tests/test_strided_array.py::test_strided_array_kernel[opencl:gpu:0]
FAILED tests/test_strided_array.py::test_strided_array_kernel[level_zero:gpu:0]

FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[opencl:gpu:0-int32] - dpctl._sycl_queue.SyclQueueCreationError: SYCL Device...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[opencl:gpu:0-int64] - dpctl._sycl_queue.SyclQueueCreationError: SYCL Device...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[opencl:gpu:0-float32] - dpctl._sycl_queue.SyclQueueCreationError: SYCL Devi...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[opencl:gpu:0-float64] - dpctl._sycl_queue.SyclQueueCreationError: SYCL Devi...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[level_zero:gpu:0-int32] - dpctl._sycl_queue.SyclQueueCreationError: SYCL De...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[level_zero:gpu:0-int64] - dpctl._sycl_queue.SyclQueueCreationError: SYCL De...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[level_zero:gpu:0-float32] - dpctl._sycl_queue.SyclQueueCreationError: SYCL ...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_dpnp_interop.py::test_consuming_array_from_dpnp[level_zero:gpu:0-float64] - dpctl._sycl_queue.SyclQueueCreationError: SYCL ...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int32-shared] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int32-device] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int32-host] - dpctl._sycl_queue.SyclQueueCreationError: SYC...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int64-shared] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int64-device] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-int64-host] - dpctl._sycl_queue.SyclQueueCreationError: SYC...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float32-shared] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float32-device] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float32-host] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float64-shared] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float64-device] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[opencl:gpu:0-float64-host] - dpctl._sycl_queue.SyclQueueCreationError: S...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int32-shared] - dpctl._sycl_queue.SyclQueueCreationErro...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int32-device] - dpctl._sycl_queue.SyclQueueCreationErro...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int32-host] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int64-shared] - dpctl._sycl_queue.SyclQueueCreationErro...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int64-device] - dpctl._sycl_queue.SyclQueueCreationErro...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-int64-host] - dpctl._sycl_queue.SyclQueueCreationError:...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float32-shared] - dpctl._sycl_queue.SyclQueueCreationEr...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float32-device] - dpctl._sycl_queue.SyclQueueCreationEr...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float32-host] - dpctl._sycl_queue.SyclQueueCreationErro...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float64-shared] - dpctl._sycl_queue.SyclQueueCreationEr...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float64-device] - dpctl._sycl_queue.SyclQueueCreationEr...
FAILED opt/conda/lib/python3.8/site-packages/numba_dppy/tests/integration/test_usm_ndarray_interop.py::test_consuming_usm_ndarray[level_zero:gpu:0-float64-host] - dpctl._sycl_queue.SyclQueueCreationErro...
```

--- a/numba_dppy/tests/integration/test_dpnp_interop.py
+++ b/numba_dppy/tests/integration/test_dpnp_interop.py
@@ -18,7 +18,7 @@ import numpy as np
 from numba import njit
 import pytest
 import numba_dppy as dppy
-from numba_dppy.tests._helper import ensure_dpnp
+from numba_dppy.tests._helper import ensure_dpnp, skip_test
 
 list_of_dtype = [
     np.int32,
@@ -47,6 +47,9 @@ def usm_type(request):
 
 def test_consuming_array_from_dpnp(offload_device, dtype):
     if not ensure_dpnp():
+        pytest.skip()
+
+    if skip_test(offload_device):
         pytest.skip()
 
     @dppy.kernel

--- a/numba_dppy/tests/integration/test_usm_ndarray_interop.py
+++ b/numba_dppy/tests/integration/test_usm_ndarray_interop.py
@@ -18,6 +18,7 @@ import numpy as np
 from numba import njit
 import pytest
 import numba_dppy as dppy
+from numba_dppy.tests._helper import skip_test
 
 import dpctl.tensor as dpt
 
@@ -47,6 +48,9 @@ def usm_type(request):
 
 
 def test_consuming_usm_ndarray(offload_device, dtype, usm_type):
+    if skip_test(offload_device):
+        pytest.skip()
+
     @dppy.kernel
     def data_parallel_sum(a, b, c):
         """

--- a/numba_dppy/tests/test_array_utils.py
+++ b/numba_dppy/tests/test_array_utils.py
@@ -20,9 +20,13 @@ import dpctl.tensor as dpt
 import dpctl.memory as dpctl_mem
 from . import _helper
 from numba_dppy.utils import has_usm_memory, as_usm_obj, copy_to_numpy_from_usm_obj
+from numba_dppy.tests._helper import skip_test
 
 
 def test_has_usm_memory(offload_device):
+    if skip_test(offload_device):
+        pytest.skip()
+
     a = np.ones(1023, dtype=np.float32)
 
     with dpctl.device_context(offload_device):
@@ -42,6 +46,9 @@ def test_has_usm_memory(offload_device):
 
 
 def test_as_usm_obj(offload_device):
+    if skip_test(offload_device):
+        pytest.skip()
+
     a = np.ones(1023, dtype=np.float32)
     b = a * 3
 

--- a/numba_dppy/tests/test_strided_array.py
+++ b/numba_dppy/tests/test_strided_array.py
@@ -29,6 +29,9 @@ def sum(a, b, c):
 
 
 def test_strided_array_kernel(offload_device):
+    if skip_test(offload_device):
+        pytest.skip()
+
     global_size = 606
     a = np.arange(global_size * 2, dtype="i4")[::2]
     b = np.arange(global_size, dtype="i4")[::-1]

